apps/api/src/routes/authRouter.ts

```diff
@@ -0,0 +1,239 @@
+ * they have successfully authenticated. The URL to this endpoint must be set in the
+ * OIDC Client as a Redirect URL.
+ *
+ * This route will validate the Authorization Code provided as a search parameter by
+ * calling the OIDC Token endpoint. This will return the authenticated user's tokens
+ * (ID, Access, refresh) and information on when these tokens will expire.
+ *
+ * Then, the access token will be used to call the OIDC User Info endpoint to retrieve
+ * the available information for this user.
+ *
+ * ALl of this information will be stored in the user session.
+ *
+ * Finally, if everything is successful, the response will redirect the user agent to
+ * the correct UI page for their role:
+ *   - Applicant: /dashboard
+ *   - Institutional Rep: ?
+ *   - DAC Member: /manage/applications
+ *
+ * If Authorization fails, the response should be a redirection to an error page. This
+ * page does not currently exist so we instead redirect to the homepage.
+ */
+authRouter.get('/token', async (req, res) => {
+	if (!authConfig.enabled) {
+		res.status(400).json({ error: 'AUTH_DISABLED', message: 'Authentication is disabled.' });
+		return;
+	}
+
+	const { code } = req.query;
+
+	try {
+		// Fetch user tokens using authorization code
+		const oauth2TokenUrl = urlJoin(authConfig.AUTH_PROVIDER_HOST, `/oauth2/token`);
+		const params = {
+			code,
+			client_id: authConfig.AUTH_CLIENT_ID,
+			client_secret: authConfig.AUTH_CLIENT_SECRET,
+			grant_type: 'authorization_code',
+			redirect_uri: getOauthRedirectUri(serverConfig.UI_HOST),
+		};
+
+		const tokenResponse = await axios.get(oauth2TokenUrl, { params });
+		const parsedTokenResponse = oidcTokenResponseSchema.safeParse(tokenResponse.data);
+		if (!parsedTokenResponse.success) {
+			logger.debug(`Token response has unexpected format.`, parsedTokenResponse.error);
+			throw new Error(`Token response has unexpected format.`);
+		}
+
+		const {
+			access_token: accessToken,
+			id_token: idToken,
+			refresh_token: refreshToken,
+			refresh_token_iat: refreshTokenIat,
+		} = parsedTokenResponse.data;
+
+		const userResponse = await axios.get(urlJoin(authConfig.AUTH_PROVIDER_HOST, `/oauth2/userinfo`), {
+			headers: { Authorization: `Bearer ${accessToken}` },
+		});
+		const parsedUserinfoResponse = oidcUserinfoResponseSchema.safeParse(userResponse.data);
+		if (!parsedUserinfoResponse.success) {
+			logger.debug(`Userinfo response has unexpected format.`, parsedUserinfoResponse.error);
+			throw new Error(`Userinfo response has unexpected format.`);
+		}
+
+		const { sub, family_name: familyName, given_name: givenName } = parsedUserinfoResponse.data;
+		// Update session with all user information retrieved from OIDC Provider
+		req.session.account = {
+			accessToken,
+			idToken,
+			refreshToken,
+			refreshTokenIat,
+		};
+		req.session.user = {
+			userId: sub,
+			familyName,
+			givenName,
+		};
+		req.session.save();
```

Comment:
This is where user session info is set. Happens after successfully confirming a login.